apps/web/src/hooks/use-locale.tsx

```diff
@@ -1,20 +1,21 @@
 import { DEFAULT_LOCALE } from '@algomart/schemas'
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
 
 export function useLocale() {
-  const [locale, setLocale] = useState<string>()
+  const [locale, setLocale] = useState(DEFAULT_LOCALE)
 
-  if (typeof window === 'undefined') {
-    return DEFAULT_LOCALE
-  } else {
-    window.addEventListener('languagechange', () => {
-      setLocale(navigator?.language)
-    })
+  useEffect(() => {
+    const callback = () => {
+      setLocale(navigator.language)
+    }
+
+    setLocale(navigator.language)
```

Comment:
Was this meant to be left outside of the callback?

Reply:
Intentionally there to set the locale right away. But might be able to avoid it? Haven't read up on that event enough to know when/how it's triggered.